### DISCUSSION TITLE
Refactor available when new rust features arrive

### DISCUSSION THREAD

**Discussion by Austaras** [swc-project/swc#4556] [General]
Currently I've found 2:
1. [if let chain](https://github.com/rust-lang/rust/issues/53667), this is imminent
2. [deref pattern](https://github.com/rust-lang/rust/issues/87121), this would bring fundamental change for swc development experience, however little progress has been seen.

**Comment by kdy1** [comment]
Wow, I really need both of them

**Comment by Austaras** [comment]
if let chain has been stabilized.

**Reply by kdy1** [reply]
Wow, great!

**Reply by kdy1** [reply]
MSRV is now the current stable rust, so we can use it.
https://github.com/swc-project/swc/pull/5856

**Reply by Austaras** [reply]
No, it has been reverted...
